homeassistant/components/jewish_calendar/services.yaml

```diff
@@ -0,0 +1,241 @@
+get_gregorian_date:
+  fields:
+    date:
+      example: 2023-01-01
+      selector:
+        date:
+    include_hebrew_date_info: &include_hebrew_date_info
+      default: true
+      selector:
+        boolean:
+    include_holiday_info: &include_holiday_info
+      selector:
+        boolean:
+    include_zmanim: &include_zmanim
+      selector:
+        boolean:
+
+get_hebrew_date:
+  fields:
+    year:
+      example: 5780
+      selector:
+        number:
+          mode: box
+    month:
+      example: Tishrei
+      selector:
+        select:
+          options:
+            - Tishrei
```

Comment:
Is there a way we can provide this list from py-libhdate? What happens if we start supporting more languages? Other spellings?
The same is especially true for holidays.

Reply:
It looks like this is possible via [`async_set_service_schema()`](https://github.com/search?q=repo%3Ahome-assistant%2Fcore%20async_set_service_schema&type=code), which is meant to register dynamically-generated services.

However, looking through its implementation, it ought to work fine to replace an already-registered service too, which means I should be able to call it to either supplement or entirely replace `services.yaml` here.

Do you think core is likely to accept that?

Reply:
It seems a bit confusing to mix services.yaml with programmatically created service schema for the same service if that's what you're talking about. Could you please elaborate a bit?

Reply:
I mean to remove service.yaml entirely and move its contents to Python code so that I can generate it dynamically.

Reply:
OK, and why is that?

Reply:
So I can automatically generate the long lists here from the underlying data in hdate.

This would also let me deduplicate the `include_*` fields that are duplicated in every service.

Reply:
Right, but how do translations work in that case?

In the PR, the list of holidays seems to be repeated thrice:
- A label
- Natural language, which seems to be a mix of Hebrew transliterated to Latin script (Chanukah) or English (Family Day)
- Natural language, which seems to be Hebrew

If you auto generate, I guess the select selector could be improved to have key-name pairs with the name according to the user configured language for holidays?

Would this work also for someone living in France for example, or that's something to improve in the future since the integration currently only allows the user to pick either "Hebrew" or "English"?

BTW, in a separate PR, you should replace this with select selector translation in strings.json: https://github.com/home-assistant/core/blob/e2dd88be6ef0792b764b0bb39374ffe0b0d511eb/homeassistant/components/jewish_calendar/config_flow.py#L47-L50